hibernate-core/src/main/java/org/hibernate/type/ComponentType.java

```diff
@@ -745,7 +751,7 @@ public Object extract(CallableStatement statement, String paramName, SharedSessi
 	}
 
 	private Object resolve(Object[] value, SharedSessionContractImplementor session) throws HibernateException {
-		return instantiator().instantiate( () -> value, session.getFactory() );
+		return instantiator( null ).instantiate( () -> value, session.getFactory() );
```

Comment:
Shouldn't you pass the component discriminator here to allow resolving the correct instantiator?

Reply:
I don't have access to the discriminator value because it's not included in the values array.

Reply:
As we discussed, this method should get the discriminator value passed.